parsl/monitoring/db_local.py

```diff
@@ -0,0 +1,218 @@
+                            if 'task_status' not in self.meta.tables.keys():
+                                task_status_table = create_task_status_table(self.meta)
+                                self.meta.create_all(con)
+                                con.execute(task_status_table.insert().values(**{k: v for k, v in info.items() if k in task_status_table.c}))
+                            # if this status table already exists, just insert the update
+                            else:
+                                task_status_table = self.meta.tables['task_status']
+                                con.execute(task_status_table.insert().values(**{k: v for k, v in info.items() if k in task_status_table.c}))
+                            return
+
+                        if 'psutil_process_pid' in info.keys():
+                            # if this is a task resource update then handle that, if the resource table DNE then create it
+                            if 'task_resources' not in self.meta.tables.keys():
+                                task_resource_table = create_task_resource_table(self.meta)
+                                self.meta.create_all(con)
+                                con.execute(task_resource_table.insert().values(**{k: v for k, v in info.items() if k in task_resource_table.c}))
+                            # if this resource table already exists, just insert the update
+                            else:
+                                task_resource_table = self.meta.tables['task_resources']
+                                con.execute(task_resource_table.insert().values(**{k: v for k, v in info.items() if k in task_resource_table.c}))
+                            return
+
+            except Exception as e:
+                print(str(e))
+                failed = True
+                time.sleep(5)
+            if not failed:
+                return
+
+
+class RemoteHandler(logging.Handler):
+    """ Handler used to pass a log to the logging server for it to be written to the database.
+    This handler is added to a logger that is returned by get_db_logger when requested. """
+    def __init__(self, web_app_host, web_app_port, request_timeout=40, retries=3, on_fail_sleep_duration=5):
+        """ Set up various behavior of the attempt to connect to the logging server. Not currently configurable. """
+        logging.Handler.__init__(self)
+        self.addr = web_app_host + ':' + str(web_app_port)
+        self.request_timeout = request_timeout
+        self.retries = retries
+        self.on_fail_sleep_duration = on_fail_sleep_duration
+
+    def emit(self, record):
+        """ Open up an HTTP connection to the logging server. This is currently blocking.
+        It should not matter for the resource monitors but could hold up the DFK logs if this connection/sending takes a lot of time. """
+        http_client = httpclient.HTTPClient()
```

Comment:
wonder about http connection sharing - does this create a new HTTP/TCP connection for every single call of `emit`? if so, does sharing an http client make that better?

Reply:
Yes, it does create a new connection with every call to emit. I am unfamiliar with connection sharing and do not know what impact it would have on performance.